### PR TITLE
[PUBDEV-4586] Make sure that performance call works for XGBoost models.

### PR DESCRIPTION
**Pull request by mmalohlava**
The change includes:
  - saving of model metrics into DKV after XGBoost model scoring
  - junit test to verify expected invariants
  - Runit, pyunit test following original bug report

### CODE REVIEW

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by navdeep-G** [comment]
LGTM

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/hex/ModelMetrics.java [hunk 1]
```diff
@@ -60,6 +58,31 @@ private void setModelAndFrameFields(Model model, Frame frame) {
     catch (Throwable t) { }
   }
 
+  public ModelMetrics withModel(Model model) {
+    _modelKey = model == null ? null : model._key;
```
Comment (michalkurka) [inline h2o-core/src/main/java/hex/ModelMetrics.java]: Maybe we should not allow overwriting not-null keys - this API could be potentially dangerous.

Perhaps an assert would be appropriate?

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
@h2o-ops please test!

**Pull request merged**